lua/core/autopairs.lua

```diff
@@ -1,51 +1,67 @@
 -- if not package.loaded['nvim-autopairs'] then
 --   return
 -- end
-local status_ok, autopairs = pcall(require, "nvim-autopairs")
-if not status_ok then
-  return
+local M = {}
+
+M.config = function()
+  O.plugin.autopairs = {
+    config = false,
+  }
 end
-local npairs = require "nvim-autopairs"
-local Rule = require "nvim-autopairs.rule"
 
--- skip it, if you use another global object
-_G.MUtils = {}
+M.setup = function()
+  if O.plugin.autopairs.config then
+    return O.plugin.autopairs.config
+  end
+
+  local status_ok, autopairs = pcall(require, "nvim-autopairs")
+  if not status_ok then
+    return
+  end
+  local npairs = require "nvim-autopairs"
+  local Rule = require "nvim-autopairs.rule"
+
+  -- skip it, if you use another global object
+  _G.MUtils = {}
 
-vim.g.completion_confirm_key = ""
-MUtils.completion_confirm = function()
-  if vim.fn.pumvisible() ~= 0 then
-    if vim.fn.complete_info()["selected"] ~= -1 then
-      return vim.fn["compe#confirm"](npairs.esc "<cr>")
+  vim.g.completion_confirm_key = ""
+  MUtils.completion_confirm = function()
+    if vim.fn.pumvisible() ~= 0 then
+      if vim.fn.complete_info()["selected"] ~= -1 then
+        return vim.fn["compe#confirm"](npairs.esc "<cr>")
```

Comment:
Old code again, shouldn't we check if `compe` is loaded first?

Reply:
The check is happening on line 42.  Line 42 will get called before line 33 because completion_confirm only gets called once the user actually enters insert mode and starts editing.